base_deterministic_session_gc/http.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Trobz <https://trobz.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import time
+import logging
+import os
+from datetime import datetime
+
+from odoo import http
+from odoo.tools import config, DEFAULT_SERVER_DATETIME_FORMAT
+
+
+_logger = logging.getLogger(__name__)
+
+
+old_session_gc = http.session_gc
+
+
+def session_gc(session_store):
+    return
+
+
+def deterministic_session_gc(session_store, session_expiry_delay=None):
+    if session_expiry_delay is None:
+        session_expiry_delay = config.get('session_expiry_delay', 60*60*24*7)
+    expired_time = time.time() - int(session_expiry_delay)
+    _logger.debug('Deleting all sessions inactive since %s',
+                  datetime.fromtimestamp(expired_time)
+                  .strftime(DEFAULT_SERVER_DATETIME_FORMAT))
+    for fname in os.listdir(session_store.path):
+        path = os.path.join(session_store.path, fname)
+        try:
+            if os.path.getmtime(path) < expired_time:
+                os.unlink(path)
+        except OSError:
+            pass
```

Comment:
why adding this except case ?
at least it should be logged no ?

Reply:
This comes straight from Odoo's original code. I guess that it was to mitigate race conditions when 2 processes would be doing the garbage collection in the same time.

2 concurrent garbage collections should not happen with this module if using the scheduled action. But nevertheless, adding logs cannot hurt, and might even help to detect access rights/disk issues in some contexts. Added! Thanks for the suggestion and the review!